engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/WarningsTest.java

```diff
@@ -148,16 +148,27 @@ private void assertWarningsForAType(Value v) {
     assertEquals("Types without and with warnings are the same", type, warningType);
     assertTrue("It is an exception. Type: " + type, warning2.isException());
     try {
-      warning2.throwException();
+      throw warning2.throwException();
     } catch (PolyglotException ex) {
       if (ex.getMessage() == null) {
         assertEquals(generator.typeError(), type);
         assertEquals(generator.typeError(), warningType);
       } else {
-        assertThat(
-            "Warning found for " + type,
-            ex.getMessage(),
-            AllOf.allOf(containsString("warn:once"), containsString("warn:twice")));
+        try {
+          assertThat(
+              "Warning found for " + type,
+              ex.getMessage(),
+              AllOf.allOf(containsString("warn:once"), containsString("warn:twice")));
+        } catch (AssertionError err) {
+          if (type != null && v.equals(warning1) && v.equals(warning2)) {
+            assertEquals(
+                "Cannot attach warnings to Error - check it is an error",
+                "Standard.Base.Error.Error",
```

Comment:
Special case to handle [DataflowErrors that cannot have Warnings](https://github.com/enso-org/enso/pull/10899#issuecomment-2317082738).